db/fmp.py

```diff
@@ -0,0 +1,168 @@
+import json
+from datetime import datetime
+
+import numpy as np
+import pandas as pd
+import pyodbc
+from tqdm import tqdm
+
+from db import DB
+from utils import edit_distance
+
+
+"""
+File Maker Pro Connection Object
+
+Desc:
+This file contains the primary logic for interacting with the file maker pro database
+"""
+
+
+class FMP(DB):
+    def __init__(self, dsn=None, username=None, password=None, connect=True):
+        DB.__init__(self)
+        if dsn is None:
+            dsn = input("Enter the FMP DSN: ")
+        if username is None:
+            username = input("Enter your FMP username: ")
+        if password is None:
+            password = input("Enter your FMP password: ")
+
+        self.__dsn = dsn
+        self.__username = username
+        self.__password = password
+
+        self.__connection_string = "DSN={0};UID={1};PWD={2}".format(
+            self.__dsn, self.__username, self.__password
+        )
+
+        # TODO: consider extracting this list to the DB class
```

Comment:
Can we make a ticket for extracting this list. I agree it should be done + would be nice to have a "central source" for the table names.

Reply:
Added issue #99